client/text.cpp

```diff
@@ -1388,42 +1389,64 @@ QString text_happiness_buildings(const struct city *pcity)
 }
 
 /**
-   Describing nationality effects that affect happiness.
+ * Describing nationality effects that affect happiness.
  */
-const QString text_happiness_nationality(const struct city *pcity)
+QString text_happiness_nationality(const struct city *pcity)
 {
-  QString str;
-  int enemies = 0;
+  if (auto effects = get_effects(EFT_ENEMY_CITIZEN_UNHAPPY_PCT);
+      !game.info.citizen_nationality || effect_list_size(effects) == 0) {
+    // Disabled in the ruleset
+    effect_list_destroy(effects);
+    return QString();
+  }
 
-  str = _("Nationality: ") + qendl();
+  auto str = QStringLiteral("<p>");
+  str +=
+      _("The presence of enemy citizens can create additional unhappiness.");
+  str += QStringLiteral(" ");
 
-  if (game.info.citizen_nationality) {
-    if (get_city_bonus(pcity, EFT_ENEMY_CITIZEN_UNHAPPY_PCT) > 0) {
-      struct player *owner = city_owner(pcity);
+  int pct = get_city_bonus(pcity, EFT_ENEMY_CITIZEN_UNHAPPY_PCT);
+  if (pct == 0) {
+    str += _("However, it is not the case in this city.");
+    return str + QStringLiteral("</p>");
+  }
 
-      citizens_foreign_iterate(pcity, pslot, nationality)
-      {
-        if (pplayers_at_war(owner, player_slot_get_player(pslot))) {
-          enemies += nationality;
-        }
-      }
-      citizens_foreign_iterate_end;
+  // This is not exactly correct, but gives a first idea.
+  int num = std::ceil(100. / pct);
+  str += QString(PL_("For every %1 citizen of an enemy nation, one citizen "
+                     "becomes unhappy.",
+                     "For every %1 citizens of an enemy nation, one citizen "
+                     "becomes unhappy.",
+                     num))
+             .arg(num);
+  str += QStringLiteral("</p><p>");
 
-      if (enemies > 0) {
-        str += QString(PL_("%1 enemy nationalist", "%1 enemy nationalists",
-                           enemies))
-                   .arg(QString::number(enemies));
-      }
+  int enemies = 0;
+  const auto owner = city_owner(pcity);
+  citizens_foreign_iterate(pcity, pslot, nationality)
+  {
+    if (pplayers_at_war(owner, player_slot_get_player(pslot))) {
+      enemies += nationality;
     }
+  }
+  citizens_foreign_iterate_end;
 
-    if (enemies == 0) {
-      str += QString(_("None."));
-    }
+  if (enemies == 0) {
+    str += _("There is <b>no enemy citizen</b> in this city.");
   } else {
-    str += QString(_("Disabled."));
+    auto unhappy = enemies * pct / 100;
+    // TRANS: Pluralized on "enemy citizen"
+    str +=
+        QString(PL_("There is %1 enemy citizen in this city, resulting in "
+                    "<b>%2 additional unhappy.</b>",
+                    "There are %1 enemy citizens in this city, resulting "
+                    "in <b>%2 additional unhappy.</b>",
+                    enemies))
+            .arg(enemies)
+            .arg(unhappy);
```

Comment:
This would be ideal as "..., resulting in %2 additional unhappy citizen." with pluralization. Could this be split on the comma and then pluralized separately?

Reply:
Splitting would assume that the preferred order is the same in all languages, which I doubt is the case... might be the lesser evil though.

Reply:
Decided to split this one that the comma. Also a candidate for #1212 though